src/librustc_typeck/check/method/probe.rs

```diff
@@ -974,8 +1004,20 @@ impl<'a, 'gcx, 'tcx> ProbeContext<'a, 'gcx, 'tcx> {
             }
         }
 
+        if let Some(uc) = unstable_candidates {
+            applicable_candidates.retain(|&(p, _)| {
+                if let stability::EvalResult::Deny { feature, .. } =
+                    self.tcx.eval_stability(p.item.def_id, ast::DUMMY_NODE_ID, self.span)
```

Comment:
At first I thought that this should be `scope_expr_id` and not `DUMMY_NODE_ID`, but it seems like this parameter only controls deprecation lints, and I guess we are issuing those elsewhere (and we wouldn't want to lint yet anyway). Do you agree?

Reply:
Agreed.